jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/config/property/PropertyConfigResolver.java

```diff
@@ -0,0 +1,162 @@
+      .build();
+  }
+
+  private HealthCheckConfiguration extractHealthCheck(HealthCheckConfiguration originalConfig, ValueProvider valueProvider) {
+    final Map<String, String> healthCheckProperties = valueProvider.getMap(ConfigKey.HEALTHCHECK, Collections.emptyMap());
+    if (healthCheckProperties != null && !healthCheckProperties.isEmpty()) {
+      final HealthCheckConfiguration config = originalConfig != null ? originalConfig : new HealthCheckConfiguration();
+      return config.toBuilder()
+        .interval(valueProvider.getString(ConfigKey.HEALTHCHECK_INTERVAL, valueOrNull(originalConfig, HealthCheckConfiguration::getInterval)))
+        .timeout(valueProvider.getString(ConfigKey.HEALTHCHECK_TIMEOUT, valueOrNull(originalConfig, HealthCheckConfiguration::getTimeout)))
+        .startPeriod(valueProvider.getString(ConfigKey.HEALTHCHECK_START_PERIOD, valueOrNull(originalConfig, HealthCheckConfiguration::getStartPeriod)))
+        .retries(valueProvider.getInteger(ConfigKey.HEALTHCHECK_RETRIES, valueOrNull(originalConfig, HealthCheckConfiguration::getRetries)))
+        .modeString(valueProvider.getString(ConfigKey.HEALTHCHECK_MODE, originalConfig == null || originalConfig.getMode() == null ? null : originalConfig.getMode().name()))
+        .cmd(extractArguments(valueProvider, ConfigKey.HEALTHCHECK_CMD, valueOrNull(originalConfig, HealthCheckConfiguration::getCmd)))
+        .build();
+    } else {
+      return originalConfig;
+    }
+  }
+
+  // Extract only the values of the port mapping
+
+  private Set<String> extractPortValues(List<String> config, ValueProvider valueProvider) {
+    final Set<String> ret = new LinkedHashSet<>();
+    final List<String> ports = valueProvider.getList(ConfigKey.PORTS, config);
+    if (ports == null) {
+      return null;
+    }
+    final List<String[]> parsedPorts = EnvUtil.splitOnLastColon(ports);
+    for (String[] port : parsedPorts) {
+      ret.add(port[1]);
+    }
+    return ret;
+  }
+
+  private Arguments extractArguments(ValueProvider valueProvider, ConfigKey configKey, Arguments alternative) {
+    return valueProvider.getObject(configKey, alternative, raw -> raw != null ? Arguments.builder().shell(raw).build() : null);
+  }
+
+  private WatchImageConfiguration extractWatchConfig(ImageConfiguration fromConfig, ValueProvider valueProvider) {
+    final WatchImageConfiguration config = fromConfig.getWatchConfiguration() != null ?
+      fromConfig.getWatchConfiguration() : new WatchImageConfiguration();
+    return config.toBuilder()
+      .interval(valueProvider.getInteger(ConfigKey.WATCH_INTERVAL, config.getIntervalRaw()))
+      .postGoal(valueProvider.getString(ConfigKey.WATCH_POSTGOAL, config.getPostGoal()))
+      .postExec(valueProvider.getString(ConfigKey.WATCH_POSTEXEC, config.getPostExec()))
+      .modeString(valueProvider.getString(ConfigKey.WATCH_POSTGOAL, config.getMode() == null ? null : config.getMode().name()))
```

Comment:
Shouldn't we be using ConfigKey.WATCH_MODE here ?

Reply:
Definitely, this seems to have been carried over from DMP:
https://github.com/fabric8io/docker-maven-plugin/blob/25e772ed663d34bf2932eb48baabd6492065b0cf/src/main/java/io/fabric8/maven/docker/config/handler/property/PropertyConfigHandler.java#L458

That should be fixed there too.

I'll change here as soon as you finish the review so I can push all the changes at once.